src/Mod/PartDesign/App/FeatureHole.cpp

```diff
@@ -1371,6 +1371,9 @@ void Hole::findClosestDesignation()
     // Intended for thread type changes
     // finds the closest diameter of the new thread type
     int threadType = ThreadType.getValue();
+    if (threadType < 0) {
+        throw Base::IndexError(QT_TRANSLATE_NOOP("Exception", "Thread type out of range"));
+    }

```

Comment:
Don't we need to check the upper bound too?

Reply:
No - `getValue()` actually does that and returns -1 if it's out of range low or high (or if the enumeration is simply invalid, though we shouldn't hit that case here).